src/SingleBookingPage/NearestBooking.js

```diff
@@ -13,194 +13,135 @@ import Notification from './bookingItem/Notification';
+      date = idx(booking, _ => _.start.time);
     }
 
-    return null;
-  };
-
-  getNearestBooking = bookings => {
-    if (!bookings || bookings.length === 0) {
-      return null;
-    }
-
-    const bookingsWithDelta = bookings
-      .map(bookingEdge => {
-        if (!bookingEdge || !bookingEdge.node) {
-          return null;
-        }
-
-        const booking = bookingEdge.node;
-        const delta = this.getDate(booking) - new Date();
-
-        return { delta, booking: booking };
-      })
-      .filter(Boolean);
-    const nearestUpcomingBooking = bookingsWithDelta
-      .filter(({ delta }) => delta > 0)
-      .reduce((nearestBooking, bookingWithDelta) => {
-        if (!nearestBooking) {
-          return bookingWithDelta;
-        }
-
-        if (nearestBooking.delta > bookingWithDelta.delta) {
-          return bookingWithDelta;
-        }
-
-        return nearestBooking;
-      }, null);
-
-    // T.O.D.O. return nearest past booking (let's wait for another GraphQL update)
-    return nearestUpcomingBooking || null;
+    return date ? new Date(date) : null;
   };
 
   render() {
-    const { bookingEdges } = this.props;
-    const nearestBooking = this.getNearestBooking(bookingEdges);
-
-    if (nearestBooking) {
-      const deltaInHours = nearestBooking.delta / 36e5; // ms to hours
-      const isUrgent = deltaInHours > 0 && deltaInHours < URGENCY_THRESHOLD;
-
-      return (
-        <div className="nearestBooking">
-          <Header booking={this.getBookingByType(nearestBooking.booking)} />
-          <Notification isUrgent={isUrgent} hoursLeft={deltaInHours} />
-          {this.renderByType(nearestBooking.booking)}
-          <div className="buttons">
-            <a
-              target="_blank"
-              rel="noopener noreferrer"
-              href={nearestBooking.booking.directAccessURL}
-            >
-              <button className="manage-booking">Manage my booking</button>
-            </a>
-          </div>
-          {isUrgent && <Contact />}
-          <style jsx>
-            {`
-              .nearestBooking {
-                width: 100%;
-                padding: 40px;
-                background-color: #f5f7f9;
-                height: 100%;
-              }
-              .buttons {
-                display: flex;
-                justify-content: center;
-                margin-top: 24px;
-              }
-              .manage-booking {
-                font-weight: bold;
-                font-size: 14px;
-                width: 260px;
-                height: 44px;
-                padding: 12px 42px;
-                border-radius: 3px;
-                background-color: #e8edf1;
-                border: none;
-                color: #46515e;
-                cursor: pointer;
-              }
-            `}
-          </style>
+    const { booking } = this.props;
+    const departure = this.getDepartureByType(booking);
+    const timeDelta = departure
```

Comment:
Could we just `if(!departure) return null` and avoid all the comparisons with `null` and this ternary here?

Reply:
Yep, you're right. 

I'm just wondering, if there's no departure, it means there's no booking to show at all - neither future not past-, isn't it? Maybe we should have a screen for that "edge" case?

Reply:
well, it's theoretical, but GraphQL can return everything but departure date, as all fields are nullable.